doc/conf.py

```diff
@@ -185,10 +185,24 @@ def use_aio():
 html_static_path = ['_static']
 
 
-## additional variables which become accessible in RST (e.g. .. ifconfig:: not no_network)
-##
+# http://stackoverflow.com/a/21449475/884770
+# http://www.sphinx-doc.org/en/stable/ext/autodoc.html#event-autodoc-skip-member
+
+# app, what, name, obj, skip, options:
+# <sphinx.application.Sphinx object at 0x2b0192ab2f90> class __module__ autobahn.asyncio.websocket True {'show-inheritance': True, 'members': <object object at 0x2b018c791710>, 'undoc-members': True}
+#
+def autodoc_skip_member(app, what, name, obj, skip, options):
+    if hasattr(obj, '__doc__') and obj.__doc__ is not None and obj.__doc__ != '':
+        if '@public' not in obj.__doc__:
+            return True
```

Comment:
Interesting!
Also, since we get the actual ``obj`` in the hook, it *could* be a real decorator (e.g. mark the function/class) or a global list somewhere like ``public_api = [ApplicationRunner, ...]`` perhaps. Neat idea, though (listing public API explicitly in code + docs).